### ISSUE TITLE
2.0.0.M1 Enables Security even if that auto configuration is excluded

### ISSUE THREAD

**Issue by Bwvolleyball** [pivotal-cf/spring-cloud-services-starters#28]
Version 2.0.0.M1 of these service connectors / starters autoconfigures Spring Security even though I have disabled it in the application.yml with the property below:

```
spring:
  autoconfigure:
    exclude: org.springframework.boot.autoconfigure.security.servlet.SecurityAutoConfiguration
```
-- edit, more details --
Just to clarify, locally security gets disabled, so the property works, as soon as the application gets deployed to PCF, security is enabled again.
-- end edit --

As an aside, why do these connectors automatically autoconfigure Spring Security? I get that the connectors themselves need Spring Security to make the handshake with PCF provided services, however, shouldn't these connectors stay out of the way of the apex application and only autoconfigure security if the apex application has included security as a dependency?

**Comment by ojhughes** [comment]
@Bwvolleyball the security auto-config dependency got included by mistake in M1. The correct dependency will be released in M2.

I don't why you are seeing different results in PCF vs local. Security can be fully disabled programatically in your app:

```
@SpringBootApplication
public class Application extends WebSecurityConfigurerAdapter {

	@Override
	protected void configure(HttpSecurity http) throws Exception {
		http.csrf().disable().authorizeRequests().anyRequest().permitAll();
	}
```

**Comment by Bwvolleyball** [comment]
Alrighty, thank you @ojhughes!

I'll go that route and then try again with M2!

I'm not sure why that was happening either, I didn't look too heavily into it, I just figured it was some sort of wiring order or property overriding when the 'cloud' profile was active.

At any rate I just wanted to bring it to your attention 😀

**Comment by ojhughes** [comment]
Thanks @Bwvolleyball 

Hopefully this fix should be in `2.0.BUILD-SNAPSHOT` so feel free to try and report back

**Comment by Bwvolleyball** [comment]
Hi @ojhughes,

I tested this again with `2.0.0.BUILD-SNAPSHOT` and it's now working as expected!

I also re-tested again with `2.0.0.M1` to make sure I wasn't crazy and the behavior reproduced 😄 

Thank you for the quick resolution!
